openmc/tallies.py

```diff
@@ -0,0 +1,3204 @@
+        Total number of filter bins accounting for all filters
+    num_bins : int
+        Total number of bins for the tally
+    shape : 3-tuple of int
+        The shape of the tally data array ordered as the number of filter bins,
+        nuclide bins and score bins
+    filter_strides : list of int
+        Stride in memory for each filter
+    num_realizations : int
+        Total number of realizations
+    with_summary : bool
+        Whether or not a Summary has been linked
+    sum : numpy.ndarray
+        An array containing the sum of each independent realization for each bin
+    sum_sq : numpy.ndarray
+        An array containing the sum of each independent realization squared for
+        each bin
+    mean : numpy.ndarray
+        An array containing the sample mean for each bin
+    std_dev : numpy.ndarray
+        An array containing the sample standard deviation for each bin
+    derived : bool
+        Whether or not the tally is derived from one or more other tallies
+    sparse : bool
+        Whether or not the tally uses SciPy's LIL sparse matrix format for
+        compressed data storage
+    derivative : openmc.TallyDerivative
+        A material perturbation derivative to apply to all scores in the tally.
+
+    """
+
+    next_id = 1
+    used_ids = set()
+
+    def __init__(self, tally_id=None, name=''):
+        # Initialize Tally class attributes
+        self.id = tally_id
+        self.name = name
+        self._filters = cv.CheckedList(_FILTER_CLASSES, 'tally filters')
+        self._nuclides = cv.CheckedList(_NUCLIDE_CLASSES, 'tally nuclides')
+        self._scores = cv.CheckedList(_SCORE_CLASSES, 'tally scores')
+        self._estimator = None
+        self._triggers = cv.CheckedList(openmc.Trigger, 'tally triggers')
+        self._derivative = None
+
+        self._num_realizations = 0
+        self._with_summary = False
+
+        self._sum = None
+        self._sum_sq = None
+        self._mean = None
+        self._std_dev = None
+        self._with_batch_statistics = False
+        self._derived = False
+        self._sparse = False
+
+        self._sp_filename = None
+        self._results_read = False
+
+    def __repr__(self):
+        string = 'Tally\n'
+        string += '{: <16}=\t{}\n'.format('\tID', self.id)
+        string += '{: <16}=\t{}\n'.format('\tName', self.name)
+
+        if self.derivative is not None:
+            string += '{: <16}=\t{}\n'.format('\tDerivative ID',
+                                              str(self.derivative.id))
+
+        filters = ', '.join(type(f).__name__ for f in self.filters)
+        string += '{: <16}=\t{}\n'.format('\tFilters', filters)
+
+        string += '{: <16}=\t'.format('\tNuclides')
+
+        for nuclide in self.nuclides:
+            string += str(nuclide) + ' '
+
+        string += '\n'
+
+        string += '{: <16}=\t{}\n'.format('\tScores', self.scores)
+        string += '{: <16}=\t{}\n'.format('\tEstimator', self.estimator)
```

Comment:
Suggestion:
we use the same format everywhere, we can just do all of this in one for loop.
this makes a bit more sense to me, since it means we are listing out all of the fields of the class.
```
nameToFieldPairs = []
nameToField.append(('ID', self.id))
nameToField.append(('Name', self.name))
nameToField.append(('ID', self.id))
nameToField.append(('ID', self.id))
...
strBuilder = []
for name, field in range:
  strBuilder = '{: <16}=\t{}\n'.format(name, field)
return '\n'.join(strBuilder)
```